pallets/author-mapping/src/lib.rs

```diff
@@ -221,14 +223,16 @@ pub mod pallet {
 			author_id: &T::AuthorId,
 			account_id: &T::AccountId,
 		) -> DispatchResult {
+			let deposit = T::DepositAmount::get();
+
+			T::DepositCurrency::reserve(&account_id, deposit)
+				.map_err(|_| Error::<T>::CannotAffordSecurityDeposit)?;
+
 			let info = RegistrationInfo {
 				account: account_id.clone(),
-				deposit: T::DepositAmount::get(),
+				deposit,
 			};
 
-			T::DepositCurrency::reserve(&account_id, T::DepositAmount::get())
-				.map_err(|_| Error::<T>::CannotAffordSecurityDeposit)?;
-
 			MappingWithDeposit::<T>::insert(&author_id, &info);
```

Comment:
this change calls `T::DepositAmount::get()` once instead of twice, but preserves everything else